lib/markdown-service.js

```diff
@@ -66,8 +66,11 @@ class MarkdownService {
             data = data.replace(/<strong>|<\/strong>/g, '**');
 
             // add escape for numbered lists
-            // https://regex101.com/r/nB8bV3/1
-            data = data.replace(/(^\d{1,2})(. )/, '$1\\\\. ');
+            // https://regex101.com/r/SO5haw/1
+            if (data.match(/^\d{1,4}\./)) {
+                // https://regex101.com/r/nB8bV3/4
+                data = data.replace(/(^\d{1,4})(.)/, '$1\\$2');
```

Comment:
As it stands, this line will match any line that starts with one to four digits followed by any non-line-breaking character.

An example where we would not want to match would be with non-numeric identifiers, such as `1a.`.

We are specifically referring to this; CURRENT:

```
data = data.replace(/(^\d{1,4})(.)/, '$1\\$2');
```

versus this; UPDATED:

```
data = data.replace(/(^\d{1,4})(\.)/, '$1\\$2');
```

With the assumption that we do not want to match non-numeric identifiers, I added a test case to the regex101 test.

https://regex101.com/r/nB8bV3/5/tests

This shows the CURRENT pattern matching, which we do not want to do, so the test fails.

Updating the regex to the UPDATED version allows the test to pass.

This can be seen here - https://regex101.com/r/nB8bV3/6/tests


What do you think?